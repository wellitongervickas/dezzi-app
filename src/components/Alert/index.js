import React from 'react';

import PropTypes from 'prop-types';
import AlertContainer from 'components/Alert/styles';

const Alert = ({ color, text }) => (
  <AlertContainer color={color}>
    <span data-testid="text">{text}</span>
  </AlertContainer>
);

Alert.defaultProps = {
  color: 'green',
};

Alert.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string.isRequired,
};

export default Alert;
