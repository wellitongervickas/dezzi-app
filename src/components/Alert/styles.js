import styled from 'styled-components';

import * as colors from 'assets/stylesheets/js/colors';
import * as sizes from 'assets/stylesheets/js/sizes';

const AlertContainer = styled.div`
  padding: ${sizes.xs()} ${sizes.sm()};
  color: ${colors.white};
  border-radius: ${sizes.xs()};
  margin-bottom: ${sizes.sm()};

  ${(props) => props.color === 'red' && `
    background-color: ${colors.red};
  `}

  ${(props) => props.color === 'green' && `
    background-color: ${colors.green};
  `}
`;

export default AlertContainer;
