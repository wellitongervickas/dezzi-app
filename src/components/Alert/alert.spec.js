import React from 'react'
import { render, cleanup } from '@testing-library/react';
import Alert from '.';

describe('Components Alert', () => {
  const text = "Warning is a warning";

  afterEach(cleanup);

  it('should have a snapshot Alert component', async () => {
    const { asFragment } = render(<Alert text={text} />);
    expect(asFragment(<Alert text={text} />)).toMatchSnapshot()
  });

  it('should have a valid alert text', async () => {
    const { getByTestId } = render(<Alert text={text} />);
    expect(getByTestId('text')).toHaveTextContent(text);
  });
});
