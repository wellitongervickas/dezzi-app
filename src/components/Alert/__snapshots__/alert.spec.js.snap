// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Alert should have a snapshot Alert component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hrgNWO"
    color="green"
  >
    <span
      data-testid="text"
    >
      Warning is a warning
    </span>
  </div>
</DocumentFragment>
`;
